### PR TITLE
Implement v2 of dashboards API

### LINKED ISSUE #3843: Implement base V2 of Dashboards
- [x] Implement Cells API (https://github.com/influxdata/chronograf/pull/3842)
- [ ] Implement Template Variables API (this work is being held off until current template variable stuff has settled)
- [ ] Implement Dashboards API (https://github.com/influxdata/chronograf/pull/3947)

### PR DESCRIPTION
**Pull request by desa**
Closes #3843 

This PR introduces a v2 of the dashboards API that consumes the cells API. It currently leaves template variables as a part of the dashboards structure. Eventually this should be removed.

  - [x] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergeable
  - [x] Tests pass
  - [x] swagger.json updated (if modified Go structs or API)

### CODE REVIEW

File: server/dashboardsv2.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+}
+
+func decodePatchDashboardRequest(ctx context.Context, r *http.Request) (*patchDashboardRequest, error) {
+	req := &patchDashboardRequest{}
+	upd := platform.DashboardUpdate{}
+	if err := json.NewDecoder(r.Body).Decode(&upd); err != nil {
+		return nil, err
+	}
+
+	req.Upd = upd
```
Comment (goller) [inline server/dashboardsv2.go]: a few extra newlines in here

File: server/swagger_v2.yml [hunk 2]
```diff
@@ -449,6 +567,93 @@ components:
           type: array
           items:
             $ref: "#/components/schemas/Cell"
+    DashboardCell:
+      type: object
+      properties:
+        X:
```
[the remark this answers is not in the document]
Comment (desa) [reply server/swagger_v2.yml]: whoopsie. That came from copying the struct 😆

File: server/swagger_v2.yml [hunk 3]
```diff
@@ -449,6 +567,93 @@ components:
...
+        W:
+          type: integer
+          format: int32
+        H:
+          type: integer
+          format: int32
+        Ref:
+          type: string
+          description: The reference to a cell from the cells API
+    TemplateValue:
```
[the remark this answers is not in the document]
Comment (desa) [reply server/swagger_v2.yml]: sure, I'll pull it out.

File: server/dashboardsv2.go [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+type getDashboardsV2Response struct {
+	Links      getDashboardsV2Links  `json:"links"`
+	Dashboards []dashboardV2Response `json:"dashboards"`
+}
+
+func (s *Service) encodeGetDashboardsResponse(w http.ResponseWriter, dashboards []*platform.Dashboard) {
+	res := getDashboardsV2Response{
+		Links: getDashboardsV2Links{
+			Self: "/chronograf/v2/dashboards",
```
[the remark this answers is not in the document]
Comment (goller) [reply server/dashboardsv2.go]: @desa that's nice because we can add `next` `prev` links; good call.

File: server/dashboardsv2.go (hunk 1 above)
Comment (desa) [reply server/dashboardsv2.go]: does this spacing work?
```go
	if err := json.NewDecoder(r.Body).Decode(&upd); err != nil {
		return nil, err
	}
	req.Upd = upd

	param := httprouter.GetParamFromContext(ctx, "id")
	req.DashboardID = platform.ID(param)
```

Comment (goller) [reply server/dashboardsv2.go]: @desa looks good to me

File: server/swagger_v2.yml [hunk 5]
```diff
@@ -123,6 +123,124 @@ paths:
...
+    summary: Get a single Dashboard
+    parameters:
+        - in: path
+          name: dashboardID
+          schema:
+            type: string
+          required: true
+          description: ID of dashboard to update
+    responses:
+        '200':
```
[the remark this answers is not in the document]
Comment (desa) [reply server/swagger_v2.yml]: fixed for cells swagger as well.

**Comment by desa** [comment]
@goller made all the changes you suggested 👍

**Pull request merged**